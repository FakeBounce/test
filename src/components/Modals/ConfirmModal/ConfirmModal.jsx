import React, { Component } from "react";
import { Button } from "reactstrap";

import { translate } from "common/methods/translations";
import Modal from "elements/Modal/Modal";
import classNames from "classnames";

class ConfirmModal extends Component {
  yesButtonClick() {
    const { handleYesButtonClick } = this.props;
    handleYesButtonClick();
  }

  noButtonClick() {
    const { handleNoButtonClick } = this.props;
    handleNoButtonClick();
  }

  redButtons() {
    return (
      <div>
        <Button color="danger" onClick={this.yesButtonClick.bind(this)}>
          {translate("globals.yes")}
        </Button>{" "}
        <Button color="secondary" onClick={this.noButtonClick.bind(this)}>
          {translate("globals.no")}
        </Button>
      </div>
    );
  }

  blueButtons() {
    return (
      <div>
        <Button color="primary" onClick={this.yesButtonClick.bind(this)}>
          {translate("globals.yes")}
        </Button>{" "}
        <Button color="secondary" onClick={this.noButtonClick.bind(this)}>
          {translate("globals.no")}
        </Button>
      </div>
    );
  }

  render() {
    const { theme } = this.props;
    const redConfirmClass = classNames("ConfirmModal", "Red");
    const blueConfirmClass = classNames("ConfirmModal", "Blue");
    if (theme === "red") {
      return (
        <Modal
          ModalFooterContent={this.redButtons.bind(this)}
          {...this.props}
          className={redConfirmClass}
        />
      );
    } else {
      return (
        <Modal
          ModalFooterContent={this.blueButtons.bind(this)}
          {...this.props}
          className={blueConfirmClass}
        />
      );
    }
  }
}

export default ConfirmModal;
