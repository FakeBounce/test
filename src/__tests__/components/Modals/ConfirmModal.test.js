import React from "react";
import { shallow, configure } from "enzyme/build";
import Adapter from "enzyme-adapter-react-16/build";
import ConfirmModal from "components/Modals/ConfirmModal/ConfirmModal";

configure({ adapter: new Adapter() });

const handleYes = jest.fn();
const handleNo = jest.fn();

const defaultTestingProps = {
  isOpen: false,
  title: "Title",
  theme: "red",
  handleYesButtonClick: handleYes,
  handleNoButtonClick: handleNo,
  toggle: () => jest.fn()
};

describe("Testing ConfirmModal component", () => {
  it("renders as expected without parameters", () => {
    const wrapper = shallow(<ConfirmModal />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders as expected with parameters (as closed)", () => {
    const wrapper = shallow(<ConfirmModal {...defaultTestingProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders as expected with parameters (as open)", () => {
    const wrapper = shallow(<ConfirmModal {...defaultTestingProps} isOpen />);
    expect(wrapper).toMatchSnapshot();
  });

  it("behave correctly when calling props", () => {
    const wrapper = shallow(<ConfirmModal {...defaultTestingProps} />);

    wrapper.instance().yesButtonClick();
    expect(handleYes.mock.calls.length).toBe(1);
    wrapper.instance().noButtonClick();
    expect(handleNo.mock.calls.length).toBe(1);
  });

  it("behave correctly when props are invalid", () => {
    const wrapper = shallow(
      <ConfirmModal
        {...defaultTestingProps}
        handleYesButtonClick={true}
        handleNoButtonClick={true}
      />
    );
    try {
      wrapper.instance().yesButtonClick();
      // Fail test if above expression doesn't throw anything.
      expect(true).toBe(false);
    } catch (e) {
      expect(e).toBeInstanceOf(TypeError);
      expect(e.message).toBe("handleYesButtonClick is not a function");
    }
    try {
      wrapper.instance().noButtonClick();
      // Fail test if above expression doesn't throw anything.
      expect(true).toBe(false);
    } catch (e) {
      expect(e).toBeInstanceOf(TypeError);
      expect(e.message).toBe("handleNoButtonClick is not a function");
    }
  });
});
