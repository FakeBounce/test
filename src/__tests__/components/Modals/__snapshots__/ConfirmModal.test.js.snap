// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ConfirmModal component renders as expected with parameters (as closed) 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConfirmModal
    handleNoButtonClick={[Function]}
    handleYesButtonClick={[Function]}
    isOpen={false}
    theme="red"
    title="Title"
    toggle={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "ModalFooterContent": [Function],
      "className": "ConfirmModal Red",
      "handleNoButtonClick": [Function],
      "handleYesButtonClick": [Function],
      "isOpen": false,
      "theme": "red",
      "title": "Title",
      "toggle": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "ModalFooterContent": [Function],
        "className": "ConfirmModal Red",
        "handleNoButtonClick": [Function],
        "handleYesButtonClick": [Function],
        "isOpen": false,
        "theme": "red",
        "title": "Title",
        "toggle": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Array [],
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Testing ConfirmModal component renders as expected with parameters (as open) 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConfirmModal
    handleNoButtonClick={[Function]}
    handleYesButtonClick={[Function]}
    isOpen={true}
    theme="red"
    title="Title"
    toggle={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "ModalFooterContent": [Function],
      "className": "ConfirmModal Red",
      "handleNoButtonClick": [Function],
      "handleYesButtonClick": [Function],
      "isOpen": true,
      "theme": "red",
      "title": "Title",
      "toggle": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "ModalFooterContent": [Function],
        "className": "ConfirmModal Red",
        "handleNoButtonClick": [Function],
        "handleYesButtonClick": [Function],
        "isOpen": true,
        "theme": "red",
        "title": "Title",
        "toggle": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Array [],
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Testing ConfirmModal component renders as expected without parameters 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConfirmModal />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "ModalFooterContent": [Function],
      "className": "ConfirmModal Blue",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "ModalFooterContent": [Function],
        "className": "ConfirmModal Blue",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Array [],
  Symbol(enzyme.__childContext__): null,
}
`;
